### PR TITLE
Add `mintlist_create` + `mintlist_add_mint_infos` instructions

### PR DESCRIPTION
**Pull request by vovacodes**
Related to #2

This creates a `MintlistAccount` struct along with instructions to:

- create / initialize the account
- add mint infos to the account

In the future, we will need to figure out:

- more instructions
  - mint NFT once the mintlist is ready
  - close the mintlist when the minting process is over (this will return the rent paid to create the mintlist)
- improve adding many NFTs at once
  - ensure that we aren't overwriting any existing NFTs
  - figure out what the maximum transaction size is

### CODE REVIEW

File: programs/nftoken/src/ix_mintlist_add_mint_infos.rs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    let mintlist_account_info = mintlist.to_account_info();
+    let mut mintlist_data = mintlist_account_info.data.borrow_mut();
+
+    let mint_info_size = MintInfo::size();
+    let insertion_byte_pos = MintlistAccount::size(mint_infos_added);
+    for (i, mint_info_arg) in mint_infos.iter().enumerate() {
+        let mint_info = MintInfo::from(mint_info_arg);
+        mint_info.serialize(&mut &mut mintlist_data[insertion_byte_pos + i * mint_info_size..])?
+        // mintlist_account.mint_infos[usize::from(mint_infos_added) + i] = mint_info.into()
```
Comment (vpontis) [inline programs/nftoken/src/ix_mintlist_add_mint_infos.rs:36]: @vovacodes are we still using this? I'm guessing it didn't work so we can remove it?

```suggestion
```

File: tests/utils/IdlCoder.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
+/**
+ * This file is copy-pasted from the anchor codebase,
+ * because it's a private API which we should either ask to make public
+ * or
```
Comment (vpontis) [inline tests/utils/IdlCoder.ts:4]:
```suggestion
 * or we can replace this with beet and our custom defined types.
```

**Review by vpontis — APPROVED** [review]
This looks great!

File: programs/nftoken/src/ix_mintlist_add_mint_infos.rs (hunk 1 above)
Comment (vovacodes) [reply programs/nftoken/src/ix_mintlist_add_mint_infos.rs:36]: Ah right, just forgot to remove it

**Pull request merged**